abi/serializer.go

```diff
@@ -154,15 +155,15 @@ func (s *serializer) doDeserialize(partsHolder *partsHolder, outputValues []any)
 	return nil
 }
 
-func (s *serializer) serializeInputOptionalValue(partsHolder *partsHolder, value InputOptionalValue) error {
+func (s *serializer) serializeOptionalValue(partsHolder *partsHolder, value *OptionalValue) error {
 	if value.Value == nil {
 		return nil
 	}
 
 	return s.doSerialize(partsHolder, []any{value.Value})
 }
 
-func (s *serializer) serializeInputMultiValue(partsHolder *partsHolder, value InputMultiValue) error {
+func (s *serializer) serializeMultiValue(partsHolder *partsHolder, value *MultiValue) error {
 	for _, item := range value.Items {
 		err := s.doSerialize(partsHolder, []any{item})
```

Comment:
Can't you do `return s.doSerialize(partsHolder, value.Items)` maybe?

Reply:
Very good point! Refactored.